sigilrecorder.lic

```diff
@@ -0,0 +1,297 @@
+		
+		#return the number of instances of roomid
+		return room_matches
+	end #check_room_data
+	
+	#attempts to navigate out of a room lacking a roomid (something that confuses go2 on which we rely) most often caused by day/night issues in a room
+	def unstuck_me(stuck_room)
+		#Use the room we think we're in to plot a path back to our starting point, if at any point along the path Room.current.id becomes recognized again, return and resume the script
+    prune_walk_list(stuck_room, @data) #add the detected bad room to nogo
+		escape_list = Map[stuck_room].path_to(@start_room)
+    current_room = stuck_room
+		escape_list.each do |roomid|
+      DRC.message("We might be in room #{current_room.to_s}.")
+      direction = Map[current_room].wayto[roomid.to_s]
+      DRC.message("Trying move command: #{direction.to_s}.")
+      move direction
+      waitrt?
+      pause 1
+      current_room = roomid
+      return if Room.current.id != nil
+    end
+    return if Room.current.id != nil
+    #If we are unable to find a mapped exit give up and exit the script
+		DRC.message('Cannot find a way out. Exiting script.')
+		exit
+	end #unstuck_me
+	
+	#uses 'perc sigil' repeatedly until both sigils are discovered
+	def find_sigils
+		room_sigils = []
+		while room_sigils.length < 2
+			#weave is a 0 RT way of detecting enemies in the room. It will auto face enemies but will not harass players or NPCs. After weaving check if the danger-room flag has tripped
+			DRC.bput('weave', /You weave back and forth in a dance-like motion/, /You must be closer to use tactical abilities on your opponent/, /There is nothing else to face/)
+			if Flags['sigil-danger-room']
+			  prune_walk_list(Room.current.id, @data) unless Room.current.id == nil
+			  return nil
+			end 
+
+			#main 'perc sigil' DRC.bput and case statement. When sigils are detected they're added to an array
+			case line = DRC.bput('perceive sigil', @sigil_list, 'Roundtime', /^You recall.*/, /^Having recently been searched/, /You are too distracted/, /Something about|This is not an appropriate place for that/)
+			when /^You recall.*/
+			  DRC.message('Room searched too recently') if @debug
+			  return nil
+			when /You are too distracted/
+			  prune_walk_list(Room.current.id, @data)
+			  return nil
+			when /Something about|This is not an appropriate place for that/
+			  prune_walk_list(Room.current.id, @data)
+			  return nil
+			when /^Having recently been searched/
+			  return nil
+			when *@sigil_list
+			  room_sigils.push(line)
+			  DRC.message("Found: #{room_sigils}") if @debug
+			end
+		end #while
+
+		#return the array of all sigils discovered in the room (should be 2 unless something went wrong)
+		return room_sigils
+	end
+	
+	#generate the list of all rooms that contain the current room's Room.current.image data
+	def region_room_list
+		data = Map.list.select do |room|
+		  room.image == Room.current.image
+		end
+
```

Comment:
This is now redundant of line 215 and the `data` variable isn't used; these lines can be removed
```suggestion
```

Reply:
Deleted redundant code